### DISCUSSION TITLE
How to serve Adonis.js behind a reverse proxy with a subpath

### DISCUSSION THREAD

**Discussion by matheusmota** [adonisjs/core#1441] [Help]
To being with, thanks for the great work with Adonis!

So, we are trying to keep multiple Adonis.js instances behind a reverse proxy (apache2) in the same domain. To achieve such separation, we have multiple locations (subpaths) pointing to different Adonis.js instances.

The mapping can be represented as follows:

```
 URL (apache2 reverse proxy) | Internal ip:port
-------------------------------------------------
 https://example.com/app1/  | 10.1.1.10:3001/
 https://example.com/app2/  | 10.1.1.10:3002/
 https://example.com/app3/  | 10.1.1.10:3003/
```

To further clarify how we are mapping, here is an excerpt of the Apache config file regarding a location:

```config
Redirect  "/app1"  "/app1/"
 <Location "/app1/>
    ProxyPreserveHost on
    Options           +Indexes +FollowSymLinks
    ProxyPass         http://172.20.0.64:10010/
    ProxyPassReverse  http://172.20.0.64:10010/
  </Location>
```

The mapping works fine.
The problem, however, is how to map the `/appX/` subpath of each Adonis.js instance. How should we configure Adonis.js to consider such subpath when using, for instance, the `{{ assetsUrl('logo.png') }}`? Is there a base_url or something similar?
We are using Adonis.js 4.1

Thanks heaps in advance. Any help will be appreciated.

**Comment by thetutlage** [comment]
For what exact reason, you want to access `appX` inside AdonisJS? Also, `Helpers.publicPath()` returns path to the `public` directory in the file system and has nothing to do with URLs

**Reply by matheusmota** [reply]
Sorry for misleading your understanding. I meant functions like the `{{ assetsUrl('logo.png') }}`. (I updated the question)

For instance, consider the usage of `{{ assetsUrl('logo.png') }}` in app1. 

The `logo.png` file is at the public folder of app1.
When rendering a page at `https://example.com/app1/level1/level2`, I am using the `{{ assetsUrl('logo.png') }}`. Therefore, I want to point to `https://example.com/app1/logo.png`, instead of `https://example.com/logo.png`.

**Reply by thetutlage** [reply]
The `assetsUrl` helper set relative paths, so ideally it shouldn't be an issue. Are you seeing any issues or are you just worried?
